lib/rules/declaration-empty-line-before/index.mjs

```diff
@@ -127,22 +128,27 @@ const rule = (primary, secondaryOptions, context) => {
 				return;
 			}
 
-			// Fix
-			if (context.fix) {
-				if (context.newline == null) return;
-
-				if (expectEmptyLineBefore) {
-					addEmptyLineBefore(decl, context.newline);
-				} else {
-					removeEmptyLinesBefore(decl, context.newline);
-				}
-
-				return;
-			}
-
 			const message = expectEmptyLineBefore ? messages.expected : messages.rejected;
+			const mutate = expectEmptyLineBefore ? addEmptyLineBefore : removeEmptyLinesBefore;
+			const semicolon = decl.parent?.raws.semicolon ? 1 : 0;
+			const endIndex = decl.toString().length + semicolon;
```

Comment:
This is the only rule that breaks the pattern of passing the length directly.

see also https://github.com/stylelint/stylelint/pull/7762/files#diff-fa27f0406ab7dfe7dfe29dfd44816f632ce1ccd274bc13ccde805c27e6ec92fbR52-R58